server/src/main/java/org/apache/druid/server/RequestLogLine.java

```diff
@@ -58,10 +78,12 @@ public String getLine(ObjectMapper objectMapper) throws JsonProcessingException
     );
   }
 
-  @JsonProperty("timestamp")
-  public DateTime getTimestamp()
+  public String getSqlQueryLine(ObjectMapper objectMapper) throws JsonProcessingException
```

Comment:
This format should be documented, but also, IMO, adjusted. Firstly, it should be on one line (this makes it easier to parse); secondly, it should be the same rough format as getNativeQueryLine (so the same parser can handle both). That means a TSV. How about doing:

```java
    return JOINER.join(
        Arrays.asList(
            timestamp,
            remoteAddr,
            objectMapper.writeValueAsString(ImmutableMap.of("queryType", "sql", "sql", sql)),
            objectMapper.writeValueAsString(queryStats)
        )
    );
```

It's sort of using a fake queryType of 'sql', which is a little weird, but makes parsing pretty easy.

Reply:
Alternatively, this:

```java
    return JOINER.join(
        Arrays.asList(
            timestamp,
            remoteAddr,
            "",
            objectMapper.writeValueAsString(queryStats),
            objectMapper.writeValueAsString(ImmutableMap.of("sql", sql))
        )
    );
```

It leaves the native-query field blank, and adds a new field on the end of the TSV for sql query. It uses a JSON object rather than emitting the SQL as-is for two reasons: (1) we can extend with more info later if we want; (2) the SQL query might have newlines and such in it, and the `objectMapper.writeValueAsString` will get rid of those.

Reply:
I agree that the format should be documented and easy to parse. Considering the two approaches, I'm in favor of the latter one because it makes it clear that SQL query is different from native query (e.g., it doesn't have `queryType`). But I also want to tweak it into
```
    return JOINER.join(
        Arrays.asList(
            timestamp,
            remoteAddr,
            "",
            objectMapper.writeValueAsString(queryStats),
            objectMapper.writeValueAsString(ImmutableMap.of("query", sql, "context", sqlQueryContext))
        )
    );
```
It uses the field naming of SqlQuery but only includes `query` and `context` fields which apply to both http and jdbc scenario. Please let me know your opinions :)